azavea/opendataphilly-ckan
# Re-enable GeoJSON previews

## Overview

Switch from ckanext-spatial extenstion to ckanext-geoview, which now contains the map previewers.
Enable the GeoJSON map previewer by default.


### Demo

![image](https://user-images.githubusercontent.com/960264/45385104-efc70e80-b5dd-11e8-9284-5b2ed4f946b4.png)

![image](https://user-images.githubusercontent.com/960264/45384036-d96b8380-b5da-11e8-998b-772ceeeb9c18.png)

![image](https://user-images.githubusercontent.com/960264/45385390-9ca18b80-b5de-11e8-97d3-9412f1530217.png)


### Notes

The `ckanext-spatial` extension we were using for the map preview feature has since been modified to move map previewing out to a separate extension, so this PR switches to that extension, `ckanext-geoview`, as we were not using any of the other `ckanext-spatial` features, such as searching by bounding box.

Previews look slightly different from the production version in that they are now displayed in a tabbed view. Enabling either of the other two previewers in the extension will add a second map preview tab to GeoJSON datasets, so here only the GeoJSON previewer is enabled, to avoid presenting multiple map preview tabs of the same dataset.

As with the current production site, if a dataset is too large to preview, either a blank map shows as the site attempts to fetch the data, or a `409 Conflict: Content is too large to be proxied` error message displays within the preview page.


## Testing Instructions

* Remove the `geojson_preview` plugin from `/etc/ckan/default/production.ini`, if present
 * `vagrant app provision`
 * Run `paster views create` in the app VM to regenerate previews (see upgrading doc for full steps)
 * Preview a (small-ish) GeoJSON dataset
 * Map should display as expected

## Checklist


Closes #161.
